From: Jeffrey Carlyle <jeff.carlyle@motorola.com>
Date: Mon, 14 Mar 2011 16:10:48 +0000 (-0500)
Subject: syslink: ipu_pm: check result of kfifo_out
X-Git-Tag: android-2.6.35-2.3-omap4.12.1~81
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=12d2398af0b33d1899455f62687eb11142d1d720

syslink: ipu_pm: check result of kfifo_out

This eliminates a compilation warning.

Change-Id: I8b79ec3f3065c8706a124c1fc4e99f43570f943c
Signed-off-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
---

Index: kernel/drivers/dsp/syslink/ipu_pm/ipu_pm.c
===================================================================
--- kernel.orig/drivers/dsp/syslink/ipu_pm/ipu_pm.c	2011-05-10 11:25:21.000000000 +0900
+++ kernel/drivers/dsp/syslink/ipu_pm/ipu_pm.c	2011-05-10 15:26:08.000000000 +0900
@@ -632,6 +632,7 @@
 	int res;
 	int rcb_num;
 	int retval;
+	int len;
 
 	if (WARN_ON(handle == NULL))
 		return;
@@ -644,9 +645,15 @@
 		/* set retval for each iteration asumming error */
 		retval = PM_UNSUPPORTED;
 		spin_lock_irq(&handle->lock);
-		kfifo_out(&handle->fifo, &im, sizeof(im));
+		len = kfifo_out(&handle->fifo, &im, sizeof(im));
 		spin_unlock_irq(&handle->lock);
 
+		if (unlikely(len != sizeof(im))) {
+			pr_err("%s: unexpected amount of data from kfifo_out: "
+					"%d\n", __func__, len);
+			continue;
+		}
+
 		/* Get the payload */
 		pm_msg.whole = im.pm_msg;
 		/* Get the rcb_num */
